src/openforms/registrations/contrib/objects_api/checks.py

```diff
@@ -0,0 +1,79 @@
+from django.utils.translation import gettext_lazy as _
+
+import requests
+from zds_client.client import ClientError
+
+from openforms.plugins.exceptions import InvalidPluginConfiguration
+from openforms.registrations.contrib.objects_api.models import ObjectsAPIConfig
+
+
+def check_config():
+    config = ObjectsAPIConfig.get_solo()
+
+    services = (
+        (
+            "objects_service",
+            "Objecten API",
+            "object",
+        ),
+        (
+            "drc_service",
+            "Documenten API",
+            "enkelvoudiginformatieobject",
+        ),
+    )
+    for service_field, api_name, api_operation in services:
+        service = getattr(config, service_field)
+        if not service:
+            raise InvalidPluginConfiguration(
+                _("{api_name} endpoint is not configured.").format(api_name=api_name)
+            )
+
+        try:
+            client = service.build_client()
+            client.list(api_operation)
+        except ClientError as e:
+            e = e.__cause__ or e
+            raise InvalidPluginConfiguration(
+                _("Client error: {exception}").format(exception=e)
+            )
+        except Exception as e:
+            raise InvalidPluginConfiguration(
+                _("Could not connect to {api_name}: {exception}").format(
+                    api_name=api_name, exception=e
+                )
+            )
+
+    urls = [
+        "objecttype",
+        "informatieobjecttype_submission_report",
+        "informatieobjecttype_submission_csv",
+        "informatieobjecttype_attachment",
+    ]
+    for service_field in urls:
+        url = getattr(config, service_field)
+        if not url:
+            # they are all defaults
+            return
+
+        try:
+            res = requests.get(url)
+            res.raise_for_status()
+        except Exception as e:
```

Comment:
as far as I understand, this makes an un-authenticated request, which will raise HTTP 403's and thus cause problems in the check, while the config would be correct, no?

The respective `client` object instead should have the correct configuration to make authenticated calls, or am I misreading this code?

Reply:
i assumed these were public but you're right and we should use the client